### PR TITLE
Additional BigQuery Options

### PR DESCRIPTION
**Pull request by stankiewicz**
this includes:

require_partition_filter
partition_expiration_days
which are common options to set in partitoned data models.
At this moment it requires additional step (manual gcloud) to patch the table.

this change is not patching/updating the existing tables, it only affect create table DDL.

### CODE REVIEW

**Comment by stankiewicz** [comment]
@canberkkoparal can I have update on this? thanks!

**Comment by canberkkoparal** [comment]
Implementation LGTM
You should add validity checking [here](https://github.com/dataform-co/dataform/blob/99ba7ace2c79561f85046929ba733f7d915856ae/core/session.ts#L736), throwing an error if target is a view or if there is no partitionBy.
Also can you add compilation tests: both [passing](https://github.com/dataform-co/dataform/blob/99ba7ace2c79561f85046929ba733f7d915856ae/tests/core/core.spec.ts#L569) and [failing](https://github.com/dataform-co/dataform/blob/99ba7ace2c79561f85046929ba733f7d915856ae/tests/core/core.spec.ts#L475)
Finally you can add even more testing by writing [integration](https://github.com/dataform-co/dataform/blob/master/tests/integration/bigquery.spec.ts) and [example](https://github.com/dataform-co/dataform/blob/master/tests/api/examples.spec.ts) tests but those are not critical.

File: protos/core.proto [hunk 1]
```diff
@@ -73,6 +73,9 @@ message BigQueryOptions {
   repeated string cluster_by = 2;
   string update_partition_filter = 3;
   map<string, string> labels = 4;
+  int32 partition_expiration_days = 5;
+  bool require_partition_filter = 6;
+
```
Comment (canberkkoparal) [inline protos/core.proto:78]: nit: delete extra empty line

**Review by canberkkoparal — CHANGES_REQUESTED** [review]

**Comment by stankiewicz** [comment]
I can't merge it (don't have permissions)  - @canberkkoparal @lewish can you merge it?

**Pull request merged**